herbert_core/classes/instrument_classes/instrument/detectors/@IX_detector_bank/IX_detector_bank.m

```diff
@@ -97,136 +97,92 @@
             if numel(varargin)==0
                 dmat = eye(3);
                 ndet0 = 1;
+            elseif numel(varargin)~=2
+                error('HERBERT:IX_detector_bank_constructor:invalid_argument', ...
+                      'must be 2 or 0 optional arguments to describe orientation type and value');
             else
```

Comment:
The comments to detector bank are similar to detector array. Incomplete serializable class. No checks for detector propety setters. No checks for property consistency. 
no constructor from structure (standard serializable constructor)

Reply:
For IX_detector_bank, the situation is different. This class was originally constructed so that the detectors would be sorted in id order. After this, any attempt to reset the fields is problematic, as there is no way to tell if the setting field matches the current sort order. Accordingly the only sensible thing to do was set all fields simultaneously with the combine field. Resetting via the constructor is probably better.

Toby has now revised this view; no sorting is now done, and the fields can now be set and checked for the purpose of instrument design. The class has now been rewritten to allow this.